web/src/components/PixiMap.tsx

```diff
@@ -147,38 +171,175 @@ const RequiresInitLayer = <T,>({
+    undefined as City | undefined
+  );
+
+  useEffect(() => {
+    setForceUpdateMap(true);
+    setLastSelectedCityDetails(
+      selectedCities.find((c) => c.id === lastSelectedCityId)
+    );
+  }, [selectedCities, lastSelectedCityId]);
+
+  useEffect(() => {
+    setForceUpdateMap(false);
+    lastSelectedCityDetails && !defaultCenter
+      ? setCenter([
+          lastSelectedCityDetails.centerLat,
+          lastSelectedCityDetails.centerLon,
+        ])
+      : undefined;
+  }, [defaultCenter, lastSelectedCityDetails]);
 
   return (
-    <Box borderRadius="lg" overflow="hidden" height={height} {...rest}>
-      <Map
-        dprs={[1, 2]}
-        provider={stamenProvider("toner-lite")}
-        attribution={stamenAttribution}
-        maxZoom={20}
-        center={center}
-        zoom={zoom}
-        onBoundsChanged={({ center, zoom }) => {
-          setCenter(center);
-          setZoom(zoom);
-        }}
-      >
-        <RequiresInitLayer useRenderer={useRenderer} options={options} />
-      </Map>
-    </Box>
+    <Stack spacing={0} position={"relative"} height={height}>
+      {showCitySelectionDropDown ? (
+        <Flex
+          position={"absolute"}
+          zIndex={5}
+          top={selectionDropdownTop}
+          left={selectionDropdownLeft}
+          background={colorMode === "light" ? "#553ACF" : "#CCC3F9"}
+          color={colorMode === "light" ? "white" : "black"}
+          boxShadow={"1px 6px 6px grey"}
+          width={"-webkit-fit-content"}
+          justifyContent={"space-between"}
+          gap={1}
+          padding={"0px 2px 0px 10px"}
+          borderRadius={"6px"}
+          alignItems={"center"}
+        >
+          <Box display={"inline"}>
+            <Text display={"inline"}>Map view</Text>
+          </Box>
+          <Center
+            height="25px"
+            margin={"2px 0px 2px 5px"}
+            background={colorMode === "light" ? "white" : "black"}
+          >
+            <Divider
+              color={colorMode === "light" ? "black" : "white"}
+              orientation="vertical"
+            />
+          </Center>
+          <Tooltip
+            isDisabled={!(lastSelectedCityId === -1 || isUpdating)}
+            label={
+              isUpdating
+                ? "Please wait while we fetch cities list"
+                : "Please select atleast 1 Location from Dashboard menu"
+            }
+          >
+            <Box display={"inline"}>
+              <Select
+                options={selectedCities.map((c) => ({
+                  label: c.name,
+                  value: c,
+                }))}
+                value={
+                  lastSelectedCityDetails
+                    ? {
+                        label: lastSelectedCityDetails.name,
+                        value: lastSelectedCityDetails,
+                      }
+                    : { label: "Select city", value: { id: -1 } }
+                }
```

Comment:
please extract these lines of code to follow our good practices https://memsql.atlassian.net/wiki/spaces/MCDB/pages/1185873978/Code+Style+Best+Practices+Guide#No-more-than-1-line-of-JavaScript-in-JSX